lib/improver/spotdata/extract_data.py

```diff
@@ -0,0 +1,374 @@
+                              units='hours')
+        return indices, bd_ids, latitude, longitude, utc_offset
+
+    def make_cube(self, cube, data, sites):
+        """
+        Construct and return a cube containing the data extracted from the
+        grids by the desired method for the sites provided.
+
+        Args:
+        -----
+        cube : iris.cube.Cube
+            The original diagnostic cube from which data has been extracted.
+
+        data : numpy.array
+            Array of diagnostic values extracted for the defined sites.
+
+        sites : dict
+            A dictionary containing the properties of spotdata sites.
+
+        Returns:
+        --------
+        cube : iris.cube.Cube
+            An irregularly (i.e. non-gridded) cube of diagnostic data extracted
+            at the spotdata sites.
+
+        """
+        latitudes = [site['latitude'] for site in sites.itervalues()]
+        longitudes = [site['longitude'] for site in sites.itervalues()]
+        utc_offsets = [site['utc_offset'] for site in sites.itervalues()]
+        site_ids = sites.keys()
+
+        indices, bd_ids, latitude, longitude, utc_offset = (
+            self._build_coordinates(
+                latitudes, longitudes, site_ids, utc_offsets))
+
+        # Add leading dimension for time.
+        data.resize(1, len(data))
+        result_cube = Cube(data,
+                           long_name=cube.name(),
+                           dim_coords_and_dims=[(cube.coord('time'), 0),
+                                                (indices, 1)],
+                           aux_coords_and_dims=[(latitude, 1),
+                                                (longitude, 1),
+                                                (utc_offset, 1),
+                                                (bd_ids, 1)],
+                           units=cube.units)
+
+        # Enables use of long_name above for any name, and then moves it
+        # to a standard name if possible.
+        result_cube.rename(cube.name())
+        return result_cube
+
+    def use_nearest(self, cube, sites, neighbours):
+        """
+        Simplest case, in which the diagnostic data value at the nearest grid
+        point, as determined by the chosen PointSelection method, is used for
+        the site.
+
+        Args:
+        -----
+        See process() above.
+
+        """
+        if (not cube.coord_dims(cube.coord(axis='y').name())[0] == 0 or
+                not cube.coord_dims(cube.coord(axis='x').name())[0] == 1):
+            raise Exception("Cube dimensions not as expected.")
+
+        data = cube.data[neighbours['i'], neighbours['j']]
+        return self.make_cube(cube, data, sites)
+
+    def orography_derived_temperature_lapse_rate(self, cube, sites, neighbours,
+                                                 orography, no_neighbours=9):
+        """
+        Crude lapse rate method that uses temperature variation and height
```

Comment:
Could you add a mention of the least-squares fitting aspect of this method to the docstring?